utils/helpers.py

```diff
@@ -1,13 +1,23 @@
 import os
 from typing import List, Optional, Dict, Tuple
 from glob import glob
+from io import BytesIO
 from numpy import void
+from dataclasses import dataclass
 
 from termcolor import colored
+from joblib import Memory
+
 from tqdm import tqdm
+from PIL import Image
+from PIL.Image import Image as PILImage
+import requests
+from tqdm.auto import tqdm
 
 from utils.constants import HOME
 
+memory = Memory(".cachedir")
```

Comment:
rather use the existing AutoML cache directory, for example `.cache/kili/automl/<project_id>/assets`